### PR TITLE
Vdeturckheim/sarif output

### PR DESCRIPTION
**Pull request by vdeturckheim**

### CODE REVIEW

**Comment by christophetd** [comment]
Let's rebase this branch on v0.1-rc?

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ guarddog pypi scan /tmp/triage.tar.gz
 guarddog pypi verify workspace/guarddog/requirements.txt
 
 # Output JSON to standard output - works for every command
-guarddog pypi scan requests --output-format=json
+guarddog pypi scan requests --json
```
Comment (vdeturckheim) [inline README.md:48]:
```suggestion
guarddog pypi scan requests --output-format=json
```

File: README.md [hunk 2]
```diff
@@ -147,7 +147,6 @@ TBD
 Authors: 
 * [Ellen Wang](https://www.linkedin.com/in/ellen-wang-4bb5961a0/)
 * [Christophe Tafani-Dereeper](https://github.com/christophetd)
```
Comment (vdeturckheim) [inline README.md:149]:
```suggestion
* [Christophe Tafani-Dereeper](https://github.com/christophetd)
* [Vladimir de Turckheim](https://www.linkedin.com/in/vladimirdeturckheim/)
```

**Comment by vdeturckheim** [comment]
@vdeturckheim add more test and remove duplicate code please

File: .github/workflows/semgrep.yml [hunk 3]
```diff
@@ -2,9 +2,13 @@ name: Semgrep scan
 
 on:
   push:
-    branches: ["main"]
+    branches:
+      - main
+      - v1.0-rc
```
Comment (christophetd) [inline .github/workflows/semgrep.yml:7]: we might want to remove this? or have a pattern like `v*`?

File: guarddog/cli.py [hunk 4]
```diff
@@ -83,11 +91,17 @@ def _verify(path, rules, exclude_rules, output_format, exit_non_zero_on_finding,
...
+        rv = js.dumps(results)
 
+    if output_format == "sarif":
+        rv = report_verify_sarif(path, list(ALL_RULES), results, ecosystem)
+
+    print(rv)
     if exit_non_zero_on_finding:
         exit_with_status_code(results)
 
+    return rv  # this is mostly for testing
```
Comment (christophetd) [inline guarddog/cli.py:103]: is it intended that we merge it? (just checking)

File: guarddog/cli.py [hunk 5]
```diff
@@ -83,11 +91,17 @@ def _verify(path, rules, exclude_rules, output_format, exit_non_zero_on_finding,
 
     if output_format == "json":
         import json as js
-        print(js.dumps(results))
+        rv = js.dumps(results)
```
Comment (christophetd) [inline guarddog/cli.py:94]: not very clear what `rv` stands for

File: guarddog/reporters/sarif.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
+import hashlib
+import json
+import yaml
+import os.path
+
+from guarddog.analyzer.metadata import get_metadata_detectors
+from guarddog.ecosystems import ECOSYSTEM
+
+
+def build_rules_help_list() -> dict:
```
Comment (christophetd) [inline guarddog/reporters/sarif.py:10]: maybe add a quick comment on what this does and the output result?

File: guarddog/reporters/sarif.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#reportingdescriptor-object
+    """
+    message = rules_documentation[rule_name] if rules_documentation[rule_name] is not None else ""
+    return {
+        "id": rule_name,
+        "defaultConfiguration": {
+            "level": "warning"
+        },
+        "shortDescription": {
+            "text": f"GuardDog rule: {rule_name}"
```
Comment (christophetd) [inline guarddog/reporters/sarif.py:74]: would it make sense (here or somewhere else in the guarddog output) to try and link to the exact rule implementation on GitHub?

File: guarddog/cli.py (hunk 4 above)
Comment (vdeturckheim) [reply guarddog/cli.py:103]: yes, this is mostly used for end-to-end testing

File: guarddog/cli.py (hunk 5 above)
Comment (vdeturckheim) [reply guarddog/cli.py:94]: it's a sqreenism 😅 , return_value, I should rename to that

File: guarddog/reporters/sarif.py (hunk 7 above)
Comment (vdeturckheim) [reply guarddog/reporters/sarif.py:74]: I think so, but this might be a stretch for now if were want to rethink the way rules are loaded in the next couple cycles

**Review by christophetd — APPROVED** [review]
